subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc

```diff
@@ -0,0 +1,137 @@
+
+[[sec:jvm_test_suite_tasks]]
+== Tasks
+
+The JVM Test Suite plugin adds the following task to the project:
+
+`test` — link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html[Test]::
+_Depends on_: `testClasses` from the `java` plugin, and all tasks which produce the test runtime classpath
++
+Runs the unit tests using JUnit by default; can be configured for other frameworks.
+
+Additional instances of link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html[Test] tasks will be automatically created for each test suite added via the `testing` extension.
+
+[[sec:jvm_test_suite_configuration]]
+== Configuration
+
+See the link:{groovyDslPath}/org.gradle.testing.base.TestingExtension.html[TestingExtension] class in the API documentation.
+
+[[sec:jvm_test_suite_terminology]]
+=== Terminology and Modeling
+The JVM Test Suite Plugin introduces some modeling concepts backed by new APIs.  Here are their definitions.
+
+==== Test Suite ====
+Each test suite is a _component_. A component is an abstract software thing. It is the description of the software to build.
+
+==== Dimensions ====
+Each test suite has _dimensions_, for example if it should be built/run on JDK8 or JDK15.
+
+==== Test Suite Target ====
+Each test suite is _realized_ from a component into a _target_. A target is an implementation of the abstract component for a particular combination of dimensions.  For the initial release of this plugin, each test suite has no configurable dimensions.  This results in a 1:1 relationship between test suite and test suite target, and a 1:1 relationship between test suite targets and a matching link:{groovyDslPath}/org.gradle.api.tasks.testing.Test.html[Test] task.
+
+Each test suite has some configuration that is common across all targets of the component:
+
+* Java version/toolchains to compile
+* Java versions/toolchains to test against
+* Testing framework
+* Sources
+* Dependencies
+
+In the future, the dimensions specified in the test suite may imply which toolchains are used to compile and run tests.
+
+Each test suite target has several things connected to it that are inherited from the component but can be configured separately as well:
+
+* Dependencies
+* Sources
+* Java version/toolchains (read-only property)
+* Testing framework (read-only property)
+
+[[sec:jvm_test_suite_examples]]
+=== Configuration Examples
+
+Here are several examples to illustrate the configurability of the `TestingExtension`.
+
+.Declaring an additional test suite
+====
+include::sample[dir="snippets/testing/test-suite-plugin/groovy",files="build.gradle[tags=configure-testing-extension]"]
+include::sample[dir="snippets/testing/test-suite-plugin/kotlin",files="build.gradle.kts[tags=configure-testing-extension]"]
+====
+
+<1> Configuration of the `TestingExtension`
+<2> Configure existing `test` suite.  This suite is automatically created for backwards compatibility.  By convention, the default `test` suite will use JUnit4.
```

Comment:
Let's not refer to this as the "default test suite", but instead the "built-in test suite". I see you called this that below which is good.

I think "default" can mean different things for different people. e.g., Some people interpret the default as the only or primary way of doing things. Or some people may expect all test suites to behave like the "default test suite".

Reply:
OK, I made sure to clean up references to "default test suite".